### PR TITLE
Crit Rework Final Submission

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Credit to Cogewerks and Goon for sprites and the base underlying idea of this system.

Alright, here's the crit rework redone.

I'd highly encourage you to read anything/everything so it's clear what has and has not changed. I'd also like to clear up some misconceptions about the intentions of the crit rework, as well. For those who are more interested in a quick and dirty of what has/has not changed, just jump to the bullet points below. **As there has been fairly dramatic alterations to the original PR, I highly recommend you read the bullet points, at least, before commenting.**

**I'd also encourage you to ask questions about how something functions, if you do not understand, as there was a lot of misunderstanding as to how things worked or how to treat in certain circumstances**

First off, the reason for the change to crit.

Current crit isn't particularly engaging; you inject a single medicine (epinephrine) then slap on a few patches/throw them into cryo and they're good to go in about 30-120 seconds (or a bit longer if they need surgery).

Likewise, for the player, your experience is largely “hit black screen and stare at it until dead”.

New crit solves both of these problems; it requires a bit more engagement on the doctors' behalf to treat  new symptoms and also forces them to not be so reliant on cryo. It also offers a bit more “wiggle room” for patients who are in crit. You can scream for help, you can try to make your way to medbay, or you have time to administer treatment to yourself. While there is still a black screen for players, eventually, it occurs much later, with this system, and at the point you would be **dead** with the old crit system; now, doctors have a chance to fight for your life.

Misconceptions I'd like to clear up:

The crit rework is absolutely not meant to be make death more meaningful, nor is it meant to push users into using any particular revive method; it's meant to be a way to alter how death, itself, and treating dying patients is handled. It offers both doctors and players a bit more wiggle room while still keeping medbay relevant and extremely important.

Things that **have changed** or been added in this crit rework:

- Crit goes through multiple stages, with more and more damage leading to accumulating negative status effects on you; these can be treated with various medications
- Three on-station species utilize the old crit system, or else they'd become unkillable (Diona, Slime, and IPC). 
  - You may have noticed Plasmamen weren't included here. They now utilize the new system with minimal changes to the species (the only change is they can now experience heart attacks)
- Other semi-playable species that use the old crit system: Skeletons, Abductors, Shadowlings, and Shadow People
- Medical Vendors have many additional medicines for treating patients in critical condition
- Cryo has been tuned downwards a bit, but it is still useful for keeping individuals in stasis
- Corazone's recipe has been removed. There's no two ways about this; this medication is completely overpowered with this new system.
- CPR has been dramatically buffed; it completely resets a patient's losebreath and dramatically heals O2 loss; it, together with epinephrine, can keep a shallow crit patient stable for a long time.
- There is a new “handheld defibrillator” . These smaller defibs do **NOT** revive, but are more effective for treating patients suffering full out cardiac arrest. They fit in medical belts and don't need to be recharged. Use these to treat; use the big defibs to bring back from the dead.
  - Any doctor's locker that has a defib will also spawn with one of these. Medical borgs also have one
- Bacon grease traitor reagent that induces immediate heart failure
- Tweaks to the upgraded robotic heart; it can stop both cardiac failure and cardiac arrest. It's also slightly less punishing if you get shocked while you have on (EMP's are still the bane of your existence, though).
- Tweaks the Health analyzer presentation, so it's easier to to see when someone is having cardiac arrest; see picture in a different post, below

Things that **haven't changed**:

- Advanced Trauma and Burn packs are still around and still in the vendors
- Strange Reagent is unchanged and functions as it currently does
- Cloning is unchanged and functions as it currently does
- Defibs are unchanged and function as they currently do, up to, and including 2 minute revivals
- You still have epipens in your survival box
- Surgery is unchanged; you still use advanced trauma packs for internal damage surgery
- There are no neurological kits in this PR; again, advanced trauma and burn kits remain

Now, for how to treat patients using this new system:

- If a patient has shock, you'll have to treat them with saline for it to go away; treating the underlying damage can help too.
- If a patient has cardiac failure, treating them epinephrine and/or atropine can remove cardiac failure
- If a patient has cardiac arrest, utilizing a handheld defib or a full defib can remove the heart attack; it's highly recommend to use the handheld defib for treating patients under cardiac arrest and not the full defibrillator; handheld defibs are specialized for treating cardiac arrest.
- Obviously, treat the underlying damage

How the new system works:

- When you lapse into critical condition you slowly accumulate O2 damage and roll for a chance to go into shock. The more damage you have, the higher a chance of you going into shock.
- When you have over 150 accumulated damage you can start rolling for a chance to acquire cardiac failure. This mostly gives you scary messages, but once it hits stage 3, it'll worsen the rate at which you acquire O2 damage and roll for a chance to put you into cardiac arrest. As with shock, the more damage you have, the higher a chance of acquire cardiac failure
- When you have over 200 damage (at this point, you'd be dead with the current implementation of crit), you roll for a chance to acquire a heart attack which makes you rapidly acquire O2 damage and brain damage.
-  At any point past 200 damage a patient can straight up die, though the chances are strongly weighted for brain damage.

Other Crap
- Fixed health analyzer sprite facing the wrong way
- Fixed some inconsistency with damage procs/direct var setting for damage
- Refactored death/crit thresholds to use a define
- Cleaned up health analyzer code ever so slightly

Changes from prior submission:
- Medical vendor has Gauze in it (unintentionally removed in prior PR submission)

:cl: Fox McCloud
add: Crit has been dramatically overhauled. Treating patients now requires a broader range of medicine. Recommend checking out a guide for how to treat using the new system!
add: Medical vendors now have styptic and silver sulfadiazine, saline, atropine, mannitol, mutadone, calomel, diphenhydramine, oculine, insulin, morphine, and potassium iodide for helping treat new patients. Amount of Gauze, Advanced Trauma, and Burn packs has also been increased.
add: Adds new handheld defibs. Utilize these for treating patients with cardiac arrest. Note: these do not revive people from the dead!
add: Doctors lockers now have a handheld defib in addition to a regular defib.
add: Medical borgs now have a handheld defib
tweak: CPR dramatically buffed. Heavily lowers O2 damage and resets losebreath to 0
tweak: Cryoxadone tuned down a little
tweak: Health analyzer UI altered ever so slightly for cardiac arerest, so it's easier to find
add: Added bacon grease reagent to traitor poison bottles; induces immediate cardiac failure
tweak: Upgraded Robotic Hearts now attempt to correct both cardiac failure and cardiac arrest
tweak; Upgraded Robotic Hearts now do not punish you quite as severely for receiving a shock while having one (EMPs will still destroy you though)
del: Corazone recipe removed
/:cl:

### CODE REVIEW

File: code/datums/diseases/food_poisoning.dm [hunk 1]
```diff
@@ -14,20 +14,12 @@
 	disease_flags = CURABLE
 	spread_flags = NON_CONTAGIOUS
 	virus_heal_resistant = TRUE
-	var/remissive = 0
 
 /datum/disease/food_poisoning/stage_act()
-	if(!remissive)
-		..()
```
Comment (Fox-McCloud) [inline code/datums/diseases/food_poisoning.dm]: FIX THIS

**Comment by farie82** [comment]
What changed compared to the previous one? I can't spot the differences myself.

File: code/modules/mob/living/carbon/human/species/vox.dm [hunk 2]
```diff
@@ -31,6 +31,7 @@
 	eyes = "vox_eyes_s"
 
 	species_traits = list(NO_SCAN, IS_WHITELISTED, NOTRANSSTING)
+	dies_at_threshold = TRUE
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/species/vox.dm]: FIX

**Comment by Fox-McCloud** [comment]
>What changed compared to the previous one? I can't spot the differences myself.

I didn't trust the map changes and others that appeared, for no reason, when I solved merge conflicts.

Aside from that, what was changed, besides that?

- Medical vendor has gauze (I accidentally removed this in the previous PR)
- Fixed typo (in both path and name) for the handheld defib (it was incorrectly `hanheld`

**Comment by datlo** [comment]
I am still eeeh on this. I suppose I will get used to it and this seems fine on the medical side of things but my time with newcrit in the few antag rounds I had didnt improve the experience. Knocking people out with raw damage seems to take forever unless you go for a decap. This might just be because I am used to damage working differently, but eeeeeeeeh.

**Comment by SkeletalElite** [comment]
Where's the health analyzer picture. Also I still think that people on cardiac arrest and cardiac failure should have unique med hud icons rather than the generic disease one.

**Comment by Sikroth** [comment]
Personally still not 100% sold on the new crit, but at this point it's mostly just personal preference. I think the changes are acceptable.

That said I still believe that there's some specific situations involving the longer death times that deserve being looked at and potentially fixed/reworked/something. For instance, terror spiders requiring their victims dead, or holoparasites remaining active after the user has been effectively downed. I know both cases will be usually just a matter of just a few more seconds, but a few more seconds being attacked by a holo or not being able to run away from a sneaky kill as a spider can have big consequences.

Just my two cents and I understand leaving these balance concerns for a later PR after the merge is done.

**Comment by KasparoVy** [comment]
Woot, great work Fox & Co.

**Comment by Fox-McCloud** [comment]
>Just my two cents and I understand leaving these balance concerns for a later PR after the merge is done.

That's the plant; it's just that we approved this and didn't want it to drag on forever and get merge conflicted to hell and back 50 more times when the system is *mostly* fine and just needs tweaks here and there.

**Pull request merged**